oaknational/Oak-Web-Application
# Feat/lesq 974/onboarding page [LESQ-974]

## Description

Music year: 2003

- add new page at /onboarding containing school picker
- refactor the school picker to make the homeschool option optional so we can exclude it on the onboarding page
- redirect to 404 page when feature flag is not enabled for user
- uses school picker components from resource pages - once we know how we're going to be using these components we should move them into a shared space and consolidate logic that is currently duplicated

## How to test

The feature flag is only active when you have added your posthog distinct_id to the rule ([see here for more details](https://www.notion.so/oaknationalacademy/I-want-a-sign-up-feature-flag-d755688e81414868908210e126de8e5a?pvs=4))

1. Go to https://deploy-preview-2659--oak-web-application.netlify.thenational.academy/onboarding
2. When the ff is not active you should be redirected to 404
3. When the ff is active you should see the onboarding page with school picker
4. The school picker on the downloads journey should be unaffected, you should still be able to select homeschool as an option and the placeholder and error messaging should reflect that https://deploy-preview-2659--oak-web-application.netlify.thenational.academy/teachers/programmes/biology-secondary-ks4-foundation-aqa/units/eukaryotic-and-prokaryotic-cells/lessons/cells/downloads?preselected=all

## Screenshots

How it should now look:
<img width="588" alt="Screenshot 2024-08-02 at 14 04 43" src="https://github.com/user-attachments/assets/2752dfc6-3720-4428-96f9-122858c58fb2">
